include/RAJA/policy/openmp/multi_reduce.hpp

```diff
@@ -0,0 +1,359 @@
+    size_t new_num_bins = container.size();
+    if (new_num_bins != m_num_bins) {
+      destroy_data(m_data, m_num_bins);
+      m_num_bins = new_num_bins;
+      m_data = create_data(container, m_num_bins);
+    } else {
+      size_t bin = 0;
+      for (auto const& value : container) {
+        m_data[bin] = value;
+        ++bin;
+      }
+    }
+  }
+
+  size_t num_bins() const { return m_num_bins; }
+
+  T identity() const { return m_identity; }
+
+  void combine(size_t bin, T const &val) { MultiReduceOp{}(m_data[bin], val); }
+
+  T get(size_t bin) const { return m_data[bin]; }
+
+private:
+  MultiReduceDataOMP const *m_parent;
+  size_t m_num_bins;
+  T m_identity;
+  T* m_data;
+
+  template < typename Container >
+  static T* create_data(Container const& container, size_t num_bins)
+  {
+    if (num_bins == size_t(0)) {
+      return nullptr;
+    }
+    auto data = RAJA::allocate_aligned_type<T>( RAJA::DATA_ALIGN, num_bins * sizeof(T) );
+    size_t bin = 0;
+    for (auto const& value : container) {
+      new(&data[bin]) T(value);
+      ++bin;
+    }
+    return data;
+  }
+
+  static void destroy_data(T*& data, size_t num_bins)
+  {
+    if (num_bins == size_t(0)) {
+      return;
+    }
+    for (size_t bin = num_bins; bin > 0; --bin) {
+      data[bin-1].~T();
+    }
+    RAJA::free_aligned(data);
+    data = nullptr;
+  }
+};
+
+/*!
+ **************************************************************************
+ *
+ * \brief  OMP multi-reduce data class template using combine on get.
+ *
+ * In this class memory is owned by each copy of the object
+ *
+ **************************************************************************
+ */
+template < typename T, typename t_MultiReduceOp >
+struct MultiReduceDataOMP<T, t_MultiReduceOp,
+    RAJA::omp::MultiReduceTuning<RAJA::omp::multi_reduce_algorithm::combine_on_get>>
+{
+  using value_type = T;
+  using MultiReduceOp = t_MultiReduceOp;
+
+  MultiReduceDataOMP() = delete;
+
+  template < typename Container,
+             std::enable_if_t<!std::is_same<Container, MultiReduceDataOMP>::value>* = nullptr >
+  MultiReduceDataOMP(Container const& container, T identity)
+      : m_parent(nullptr)
+      , m_max_threads(omp_get_max_threads())
+      , m_num_bins(container.size())
+      , m_padded_threads(pad_threads(m_max_threads))
+      , m_padded_bins(pad_bins(m_num_bins))
+      , m_identity(identity)
+      , m_data(nullptr)
+  {
+    m_data = create_data(container, identity, m_num_bins, m_max_threads, m_padded_bins, m_padded_threads);
+  }
+
+  MultiReduceDataOMP(MultiReduceDataOMP const &other)
+      : m_parent(other.m_parent ? other.m_parent : &other)
+      , m_num_bins(other.m_num_bins)
+      , m_padded_threads(other.m_padded_threads)
+      , m_padded_bins(other.m_padded_bins)
+      , m_identity(other.m_identity)
+      , m_data(other.m_data)
+  { }
+
+  MultiReduceDataOMP(MultiReduceDataOMP &&) = delete;
+  MultiReduceDataOMP& operator=(MultiReduceDataOMP const&) = delete;
```

Comment:
Can you provide some context as to why copy construction is defined but not copy assignment?

Reply:
Copy construction is necessary to copy it into the lambda.
Copy assignment would make managing the lifetimes of resources used in the reducer more complicated.